### PR TITLE
RPi: Stream all formats when flashing sdcard

### PR DESCRIPTION
**Pull request by asdil12**
This will allow larger images that would exceed worker RAM.

### CODE REVIEW

File: data/generalhw_scripts/flash_sd_rootless.sh [hunk 1]
```diff
@@ -25,38 +25,25 @@ hdd_size=$4
...
-	dd if=$uncompressed_full_path of=/dev/disk/by-id/usb-LinuxAut_sdmux_HS-SD_MMC_${device_serial}-0:0 oflag=sync bs=8M
+	image_to_flash_extension="${image_to_flash##*.}"
+	if [ "$image_to_flash_extension" == "qcow2" ] ; then
+		qemu-img dd -f qcow2 -O raw if=$image_to_flash of=$device_link bs=8M
+	elif [ "$image_to_flash_extension" == "xz" ] ; then
+		xzcat --threads=0 $image_to_flash | dd of=$device_link oflag=sync bs=8M
+	elif [ "$image_to_flash_extension" == "gz" ] ; then
+		zcat $image_to_flash | dd of=$device_link oflag=sync bs=8M
+	else
+		cat $image_to_flash | dd of=$device_link oflag=sync bs=8M
```
Comment (ggardet) [inline data/generalhw_scripts/flash_sd_rootless.sh:45]: Is there an advantage/disadvantage over  `dd if=$image_to_flash of=$device_link oflag=sync bs=8M` ?

File: data/generalhw_scripts/flash_sd_rootless.sh [hunk 2]
```diff
@@ -13,7 +13,7 @@ if [ "$#" -ne 4 ]; then
     exit 1;
 fi
 
-# Get destination: /storage/
+# Get destination (tmpdir to extract image): /storage/
 destination_folder=$1
```
Comment (ggardet) [inline data/generalhw_scripts/flash_sd_rootless.sh:17]: `destination_folder` is not used anymore apparently, and thus, could be removed.

**Comment by ggardet** [comment]
> This will allow larger images that would exceed worker RAM.

That would exceed worker **disk space**, not RAM

**Comment by asdil12** [comment]
> > This will allow larger images that would exceed worker RAM.
> 
> That would exceed worker **disk space**, not RAM

On my setup the destination dir is `/dev/shm/…` to save sdcard write cycles - so that's why I run out of RAM.
But yes it will also help with disk space.

File: data/generalhw_scripts/flash_sd_rootless.sh (hunk 2 above)
Comment (asdil12) [reply data/generalhw_scripts/flash_sd_rootless.sh]: Right. Updated script and docs.

**Pull request merged**